addon/utils.js

```diff
@@ -150,3 +150,73 @@ export function recursiveObjectCreate (object) {
 
   return object
 }
+
+/**
+ * mine an object for a value
+ * @param {Object} obj the object to mine
+ * @param {String} valuePath the path to the value in the object
+ * @returns {String} the value
+ */
+export function findValue (obj, valuePath, startPath = '') {
+  const depths = startPath.split('.')
+  const valueLevels = valuePath.split('./')
+  const parentLevels = valueLevels.filter((element) => {
+    return element === '.'
+  })
+  const valueKey = valueLevels.pop()
+  const absValuePath = _.without(depths.slice(0, depths.length - parentLevels.length - 1), '', '.').join('.')
+  const absValueKey = [absValuePath, valueKey].join('.')
+  return _.get(obj, absValueKey)
+}
+
+/**
+ * populates values in an Orchestrate-style 'q' or 'p' queryParam valuePath
+ * @param {Object} valueObj the value object to mine for query values
+ * @param {String} queryValue the queryParam value to parse
+ * @returns {String} the populate queryValue
+ */
+export function parseOrchFilters (valueObj, queryValue, startPath = '') {
+  const newQueryFilters = queryValue.split(',').map((param) => {
+    const pieces = param.split(':')
+    return `${pieces[0]}:${parseOrchFilterVariables(valueObj, pieces[1], startPath)}`
+  })
+  return newQueryFilters.join(',')
+}
+
+/**
+ * finds variables in orch-style queryParam values
+ * @param {Object} valueObj the value object to mine for query values
+ * @param {String} orchFilter the filter to parse
+ * @returns {String} the populated filter
+ */
+export function parseOrchFilterVariables (valueObj, orchFilter, startPath = '') {
+  let result = orchFilter
+  if (orchFilter.indexOf('${') !== -1) {
+    const valueVariable = orchFilter.split('${')[1].split('}')[0]
+    result = findValue(valueObj, valueVariable, startPath)
+  }
+  return result
+}
+
+/**
+ * grooms the query for variables and stitches it back together
+ * @param {Object} valueObj the value object to mine for values
+ * @param {Object} query the definition from the model schema
+ * @returns {Object} the populated query
+ */
+export function createOrchQuery (valueObj, query, startPath = '') {
+  const keys = _.keys(query)
+  const populatedQuery = {}
+  _.each(keys, (key) => {
+    // get each key/value pair
+    switch (key) {
+      case 'q':
+      case 'p':
+        populatedQuery[key] = parseOrchFilters(valueObj, query[key], startPath)
+        break
+      default:
+        populatedQuery[key] = query[key]
+    }
```

Comment:
Yeah, this doesn't seem right. So, if I'm understanding this code right, the only place where one can use variable substitution in the queries is inside a `q` or `p` query. What if you want to pull another query param from somewhere else, like `resourceTypeId` in an orch API, or any query param in a non orch API? I think if you look for the `${}` pattern in *all* parts of the query and substitute it, it will make it much more useful and powerful. I could imagine even wanting to pull the parameter *name* from somewhere else (a stretch, I know but doable if you just look for and replace `${}` instead of enforcing that it only exist at the right side of a `:` in an orch-style API request.